test/jasmine/tests/geo_test.js

```diff
@@ -650,39 +650,93 @@ describe('Test Geo layout defaults', function() {
             });
         }
 
-        it('- base case', function() {
-            layoutIn = {
-                geo: { visible: false }
-            };
+        [true, false, undefined].forEach(function(q) {
+            it('- base case | ' + q, function() {
+                layoutIn = {
+                    template: {
+                        layout: {
+                            geo: {
```

Comment:
sorry to be a pain, but I still don't see a test case that sets `template.layout.geo.visible=false`, does NOT set `layout.geo.visible`, and verifies that `template.layout.geo.show* settings are still honored in `fullLayout.geo`

Reply:
Added in https://github.com/plotly/plotly.js/pull/4483/commits/9448c8f6b6d968d76e6a49b4bb92a78ecb783b99.